### PR TITLE
Some refactoring

### PR DESCRIPTION
**Pull request by mohammja**
-Documenting all controllers classes
-Refactoring all the requests.

### CODE REVIEW

File: Backend/src/main/java/com/harmonia/backend/controller/DirectMessageController.java [hunk 1]
```diff
@@ -51,6 +87,12 @@ public ResponseEntity<DirectMessage> sendDMessage(@RequestBody DMessageRequest d
...
 
+    /**
+     * Handles DELETE requests to delete a direct message.
+     *
+     * @param directMessage a DirectMessage object representing the direct message to delete
+     * @return a ResponseEntity containing a message indicating that the direct message was deleted, or a BAD_REQUEST status if an error occurred
+     */
     @DeleteMapping
     @CrossOrigin
     public ResponseEntity<String> deleteDMessage(@RequestBody DirectMessage directMessage) {
```
Comment (psolizrodriguez) [inline Backend/src/main/java/com/harmonia/backend/controller/DirectMessageController.java:98]: @DeleteMapping("dmessageId/{dmessageId}")
Deletes are always done by the primary key.

File: Backend/src/main/java/com/harmonia/backend/controller/PublicMessageController.java [hunk 2]
```diff
@@ -8,38 +8,74 @@
...
 
+    /**
+     * Returns a list of all public messages.
+     *
+     * @param dmId (optional) the id of the public message to filter by.
+     * @return an iterable of PublicMessage objects.
+     */
     @GetMapping
     @CrossOrigin
     public Iterable<PublicMessage> listMessages(@RequestParam(name = "pmessageId", required = false) String dmId) {
```
Comment (psolizrodriguez) [inline Backend/src/main/java/com/harmonia/backend/controller/PublicMessageController.java:41]: Unless you want to do the getById and the list all in the same endpoint, I would remove the @RequestParam(name = "pmessageId", required = false) String dmId or give it a purpose like channelId

File: Backend/src/main/java/com/harmonia/backend/controller/ServerController.java [hunk 3]
```diff
@@ -7,60 +7,114 @@
...
     }
 
+
+    /**
+     * Returns a list of servers filtered by serverCategory.
+     *
+     * @param serverCategory the category of the server to filter by
+     * @return a list of servers with matching serverCategory
+     */
     @GetMapping("category/{serverCategory}")
```
Comment (psolizrodriguez) [inline Backend/src/main/java/com/harmonia/backend/controller/ServerController.java:53]: maybe use  @GetMapping("serverCategory/{serverCategory}") to make it more standard

File: Backend/src/main/java/com/harmonia/backend/controller/UserController.java [hunk 4]
```diff
@@ -9,18 +9,38 @@
...
 
+    /**
+     * Get a list of all users or a list of users by username if a username is provided
+     *
+     * @param userName (optional) the username to search for
+     * @return an iterable list of UserResponse objects
+     */
     @GetMapping
     @CrossOrigin
     public Iterable<UserResponse> listUsers(@RequestParam(name = "username", required = false) String userName) {
```
Comment (psolizrodriguez) [inline Backend/src/main/java/com/harmonia/backend/controller/UserController.java:46]: You already have @GetMapping("username/{username}") a few lines above, remove @RequestParam(name = "username", required = false) String userName

File: Backend/src/main/java/com/harmonia/backend/controller/UserController.java [hunk 5]
```diff
@@ -30,24 +50,48 @@ public Iterable<UserResponse> listUsers(@RequestParam(name = "username", require
...
 
 
+    /**
+     * Authenticate a user by their username and password
+     *
+     * @param credentials a Map containing the username and password of the user
+     * @return a ResponseEntity with a UserResponse object if the user is authenticated, or a not found response if not
+     */
     @PostMapping("login")
     public ResponseEntity<UserResponse> loginUser(@RequestBody Map<String, String> credentials) {
```
Comment (psolizrodriguez) [inline Backend/src/main/java/com/harmonia/backend/controller/UserController.java:96]: Maybe replace the (@RequestBody Map<String, String> credentials) for a DTO  UserCredentialsPO

File: Backend/src/main/java/com/harmonia/backend/controller/UserController.java [hunk 6]
```diff
@@ -59,21 +103,42 @@ public ResponseEntity<UserResponse> loginUser(@RequestBody Map<String, String> c
...
     }
 
-    @DeleteMapping("/{userId}")
+    /**
+     * Delete a user by their user ID
+     *
+     * @param userId the ID of the user to delete
+     * @return a ResponseEntity with a status of OK
+     */
+    @DeleteMapping("/userId/{userId}")
```
Comment (psolizrodriguez) [inline Backend/src/main/java/com/harmonia/backend/controller/UserController.java:112]: Remove extra "/" from mapping @DeleteMapping("/userId/{userId}")

File: Backend/src/main/java/com/harmonia/backend/controller/UserController.java [hunk 7]
```diff
@@ -59,21 +103,42 @@ public ResponseEntity<UserResponse> loginUser(@RequestBody Map<String, String> c
...
 
-    @PutMapping("/{userId}")
+    /**
+     * Edit a user by their user ID
+     *
+     * @param userId the ID of the user to edit
+     * @param user   the User object containing the updated user data
+     * @return a ResponseEntity with a UserResponse object and a status of OK
+     */
+    @PutMapping("/userId/{userId}")
```
Comment (psolizrodriguez) [inline Backend/src/main/java/com/harmonia/backend/controller/UserController.java:127]: Remove extra "/" from  @PutMapping("/userId/{userId}")

File: Backend/src/main/java/com/harmonia/backend/controller/UserController.java [hunk 8]
```diff
@@ -59,21 +103,42 @@ public ResponseEntity<UserResponse> loginUser(@RequestBody Map<String, String> c
...
-    @PutMapping("/{userId}/password")
+    /**
+     * Changes the password of the user identified by userId
+     *
+     * @param userId                the ID of the user to change the password for
+     * @param changePasswordRequest the new password to set for the user
+     * @return a ResponseEntity with a UserResponse and HTTP status OK if the password was changed successfully,
+     * or HTTP status NOT_FOUND if the user is not found
+     */
+    @PutMapping("/userId/{userId}/password")
```
Comment (psolizrodriguez) [inline Backend/src/main/java/com/harmonia/backend/controller/UserController.java:141]: Remove extra "/" from  @PutMapping("/userId/{userId}/password")

File: Backend/src/main/java/com/harmonia/backend/controller/PublicMessageController.java [hunk 9]
```diff
@@ -8,38 +8,74 @@
...
 
-    @PutMapping("/{messageId}")
+    /**
+     * Updates an existing public message.
+     *
+     * @param messageId     the id of the public message to be updated.
+     * @param publicMessage the PublicMessage object containing the updated message text.
+     * @return a ResponseEntity with a message indicating success or failure of the update.
+     */
+    @PutMapping("/messageId/{messageId}")
```
Comment (psolizrodriguez) [inline Backend/src/main/java/com/harmonia/backend/controller/PublicMessageController.java:78]: Remove extra "/" from  @PutMapping("/messageId/{messageId}")

File: Backend/src/main/java/com/harmonia/backend/controller/ChannelController.java [hunk 10]
```diff
@@ -60,7 +60,7 @@ public ResponseEntity<Void> deleteChannel(@PathVariable Long channelId) {
      * @param channel the updated Channel entity.
      * @return a ResponseEntity with a success message and HTTP status code.
      */
-    @PutMapping("/{channelId}")
+    @PutMapping("/channelId/{channelId}")
```
Comment (psolizrodriguez) [inline Backend/src/main/java/com/harmonia/backend/controller/ChannelController.java:63]: Remove extra "/" from @PutMapping("/channelId/{channelId}")

**Review by psolizrodriguez — CHANGES_REQUESTED** [review]
Good job adding comments, fix the Mappings in the controllers please

**Review by Jihau — APPROVED** [review]

**Pull request merged**